docs/DeveloperGuide.md

```diff
@@ -1,38 +1,360 @@
-# Developer Guide
+# Modganiser Developer Guide
+
+## Introduction
+
+---
+
+Modganiser is **effortless module planning, at your fingertips via the Command Line Interface** (CLI). It helps Information security students graduate in time by giving you a platform to plan all 4 years of your modules to ensure that you meet graduation requirements (UE/GEs).  If you can type fast, Modganiser can get your schedule up  faster than traditional GUI apps.
+
+## Table of Contents
+
+---
+
+[Introduction](#introduction)
+
+[Table of contents](#table-of-contents) 
+0. [Acknowledgements](#acknowledgements)
+1. [Setting Up](#setting-up)
+2. [Architecture](#architecture)
+   + 2.1. [Storage](#1-storage-component)
+   + 2.2. [Parser](#2-parser-component)
+   + 2.3. [ModuleList](#3-modulelist-component)
+   + 2.4. [Print](#4-print-component)
+3. [Design and Implementation](#3-design-and-implementation)
+   + 3.1. [ManPage](#31-manpage)
+   + 3.2. [Parser](#32-parser)
+   + 3.3. [Add Modules](#33-add-modules)
+   + 3.4. [Find Modules](#34-find-modules)
+   + 3.5. [Delete Modules](#35-delete-modules)
+   + 3.6. [Edit Modules](#36-edit-modules)
+   + 3.7. [Add or Update Grade](#37-add-or-update-grade)
+   + 3.8. [Calculate CAP](#38-calculate-cap)
+4. [Appendix: Requirements](#appendix--requirements)
+   + 4.1. [Product Scope](#product-scope)
+   + 4.2. [Target User Profile](#target-user-profile)
+   + 4.3. [User Stories](#user-stories)
+   + 4.4. [Non-Functional Requirements](#non-functional-requirements)
+   + 4.5. [Glossary](#glossary)
+5. [Appendix: Instructions for manual testing](#appendix--instructions-for-manual-testing)
+
+
+## Introduction 
+
 
```

Comment:
There is both introduction before table of contents, it bay be good to only have one